### ISSUE TITLE
EF Core SqlClient and App Insights

### ISSUE THREAD

**Issue by MaximBalaganskiy** [dotnet/corefx#16176]
What can be the reason for not seeing raw SQL in EF app insights in net core project? Not locally nor in Azure. 

I remember at some point with one of the RCs I saw SQL logged but not anymore.
Originally, I posted the question here https://github.com/aspnet/EntityFramework/issues/5510 but was referred to this repo.

**Comment by karelz** [comment]
What you don't see? Which type of project? (is EF special project type?)
Do you have minimal repro? (starting with new app without anything)

**Comment by MaximBalaganskiy** [comment]
EF is EntityFramework, used in Asp.Net Core 1.1 project. Dependency telemetry does not have SQL text.

**Comment by saurabh500** [comment]
@MaximBalaganskiy What kind of telemetry are you expecting here? 
From what I know SqlClient doesn't emit any SQL on its own. There is a tracing mechanism on .Net Framework which is disabled by default and is usually not enabled without some registry modifications and some setup on the client machine.. 
I don't think this is a SqlClient issue. 

Are you contrasting the usecase with .Net Desktop? If yes, can you give me the steps to follow to get the SQL logging you are talking about?

**Comment by MaximBalaganskiy** [comment]
Here is the comment which got me here 
https://github.com/aspnet/EntityFramework/issues/5510#issuecomment-230902144

**Comment by MaximBalaganskiy** [comment]
I used to see the actual SQL command in dependency details in VS
![image](https://cloud.githubusercontent.com/assets/3712700/24130688/432b3f22-0e3e-11e7-9e7a-c7e87d866667.png)
and in Azure. But now I cannot even reproduce this with EF6

**Comment by rwatjen** [comment]
This seems to still not work with ASP.NET Core 2.0 and Entity Framework Core 2.0. The Azure Application Insights team points here (the link that @MaximBalaganskiy has posted). It would be great if someone could take on the issue and either solve it or make it very clear in the Application Insights documentation that this does not actually work for Core.

**Comment by rwatjen** [comment]
I was just informed by @cijothomas that this works with Application Insights for .NET Core 2.2.0-beta1. (https://github.com/Microsoft/ApplicationInsights-aspnetcore/releases/tag/v2.2.0-beta1)

**Comment by cijothomas** [comment]
Yes. If you .net core application is targeting the full .NET Framework, then sql calls should have been automatically monitored.
But if you are targeting .NET Core, then support for this started with 2.2.0-beta1 of the SDK.
https://github.com/Microsoft/ApplicationInsights-aspnetcore/releases/tag/v2.2.0-beta1
You need to explicitly update to the above package, as the version of ApplicationInsights SDK shipped with Microsoft.AspNetCore.All 2.0 is 2.1.1.
